### ISSUE TITLE
posterior_predict gives incorrect results for less than full dataset as `newdata`

### ISSUE THREAD

**Issue by apollostream** [stan-dev/rstanarm#73]
While working through the examples in the vingette ["How to use rstanarm"](https://cran.r-project.org/web/packages/rstanarm/vignettes/rstanarm.html), I found what appears to be a bug in function `posterior_predict` when it executes the code given in "Step 4: Analyze manipulations of predictors." 

The results are incorrect -- and therefore, the accompanying text stating "As can be seen, two women — one with a high school degree and another with a college degree — are expected to have about the same probability of agreeing with the survey question." is in error.

I've attached a zipped R script that reproduces the error, which happens when setting the `newdata=` option to the cases for which we'd like predictions:  [demo_rstanarm.zip](https://github.com/stan-dev/rstanarm/files/120823/demo_rstanarm.zip).  It also provides a means of getting the correct results by sending in the entire original `womensrole` data.frame as `newdata` and focusing upon just the two columns of interest in the resulting matrix.  Go to the end of the script to see the section that demonstrates the bug.

I've also attached the PDF file generated by that R script as a notebook:  [demo_rstanarm.pdf](https://github.com/stan-dev/rstanarm/files/120822/demo_rstanarm.pdf). See the section at hyperlink "Demonstrating the bug in posterior_predict".

**Comment by jgabry** [comment]
Thanks for reporting this. Are you using the CRAN version? There's a decent
chance this was fixed recently on Github and will be in the next release.
If you want to try with the development version the instructions for
installing are in the readme file.

We'll also take a look through the attachments you shared (thanks for
sending all of that) and make sure it's already fixed. If not we'll get a
fix in for the next release.

**Comment by jgabry** [comment]
Ok, so the `posterior_predict` issue had already fixed on GitHub (https://github.com/stan-dev/rstanarm/commit/52608d1e9e525035186acab9946d6f73714e5777), but there were some issues with the code in the vignette that we need to resolve. Those should now be fixed on the master branch (https://github.com/stan-dev/rstanarm/commit/eaac7832c9f9ffb58a4216580799ad6126461819, https://github.com/stan-dev/rstanarm/commit/3b76bf995ec4d1a0de17f084b1d51f0dc09f0d16). Thanks again for reporting this.

**Comment by apollostream** [comment]
Thanks for the quick response. I downloaded the development version and tested my script. It DID work correctly upon modification to include `agree` and `disagree` values in the `newdata` data.frame.  I also see that the vignette included in the development version has also been edited to account for the need to include the `agree` and `disagree` values in the `newdata`, too. And the text has been changed to acknowledge the significant difference in females with 12 years of education vs. those with 16 years.
Congratulations on rstanarm: it is just an excellent piece of work!
